// umi-plugin-react-native/dist/react-native-web.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.requireNativeComponent = exports.TextInput = exports.Image = exports.Text = exports.ART = exports.ViewPropTypes = void 0;
const tslib_1 = require("tslib");
// ref to https://levelup.gitconnected.com/react-native-typescript-and-react-native-web-an-arduous-but-rewarding-journey-8f46090ca56b
// https://github.com/reactrondev/react-native-web-swiper/issues/41
const react_native_web_1 = require("react-native-web");
// Let's export everything from react-native-web
tslib_1.__exportStar(require("react-native-web"), exports);
// Improve API compatibility with React Native by exporting 'react-art' as 'ART'.
//https://github.com/necolas/react-native-web/commit/321051b72366666ba6e28985cff017cb6d3c506d
const react_art_1 = tslib_1.__importDefault(require("react-art"));
// And let's stub out everything that's missing!
exports.ViewPropTypes = {
    style: () => { },
};
react_native_web_1.TextInput.propTypes = {
    style: () => { },
};
react_native_web_1.Text.propTypes = {
    style: () => { },
};
react_native_web_1.Image.propTypes = {
    style: () => { },
    source: () => { },
};
exports.ART = react_art_1.default;
exports.Text = react_native_web_1.Text;
exports.Image = react_native_web_1.Image;
exports.TextInput = react_native_web_1.TextInput;
const requireNativeComponent = () => { };
exports.requireNativeComponent = requireNativeComponent;
