// dumi-theme-mobile/es/components/Device.js
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { if (typeof Symbol === "undefined" || !(Symbol.iterator in Object(arr))) return; var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

import React, { useState, useContext, useEffect } from 'react';
import QRCode from 'qrcode.react';
import { context, usePrefersColor } from 'dumi/theme';
import './Device.less';

var Device = function Device(_ref) {
  var url = _ref.url,
      className = _ref.className;

  var _useState = useState(Math.random()),
      _useState2 = _slicedToArray(_useState, 2),
      renderKey = _useState2[0],
      setRenderKey = _useState2[1];

  var _usePrefersColor = usePrefersColor(),
      _usePrefersColor2 = _slicedToArray(_usePrefersColor, 1),
      color = _usePrefersColor2[0];

  var _useContext = useContext(context),
      mode = _useContext.config.mode; // re-render iframe if prefers color changed


  useEffect(function () {
    setRenderKey(Math.random());
  }, [color]);
  return /*#__PURE__*/React.createElement("div", {
    className: ['__dumi-default-device'].concat(className).join(' '),
    "data-device-type": "iOS",
    "data-mode": mode
  }, /*#__PURE__*/React.createElement("div", {
    className: "__dumi-default-device-status"
  }, /*#__PURE__*/React.createElement("span", null, "dumi"), /*#__PURE__*/React.createElement("span", null, "10:24")), /*#__PURE__*/React.createElement("iframe", {
    title: "dumi-previewer",
    src: url,
    key: renderKey
  }), /*#__PURE__*/React.createElement("div", {
    className: "__dumi-default-device-action"
  }, /*#__PURE__*/React.createElement("button", {
    className: "__dumi-default-icon",
    role: "refresh",
    onClick: function onClick() {
      return setRenderKey(Math.random());
    }
  }), /*#__PURE__*/React.createElement("button", {
    className: "__dumi-default-icon",
    role: "qrcode"
  }, /*#__PURE__*/React.createElement(QRCode, {
    value: url,
    size: 96
  })), /*#__PURE__*/React.createElement("a", {
    href: url,
    target: "_blank",
    rel: "noreferrer",
    className: "__dumi-default-icon",
    role: "open-demo"
  })));
};

export default Device;
